harmonica/_io/geosoft_gxf_io.py

```diff
@@ -0,0 +1,352 @@
+
+"""
+
+"""
+Function to read USGS GXF file
+"""
+
+import numpy as np
+import xarray as xr
+from typing import Tuple, List, Dict, Union, Any
+
+def read_gxf_raw(infile: str) -> Tuple[List[str], Dict[str, str]]:
+    """
+    Read a GXF file and return raw data list and headers.
+    Following official GXF specifications from Geosoft.
+    
+    Parameters:
+    infile (str): Path to the GXF file
+    
+    Returns:
+    tuple: (data_list, headers)
+        - data_list: list of raw data strings
+        - headers: dictionary of GXF headers
+    """
+    # Read entire file
+    with open(infile) as f:
+        lines = [line.strip() for line in f.readlines()]
+    
+    # Create dictionary with headers and parameters
+    headers: Dict[str, str] = {}
+    data_list: List[str] = []
+    reading_data = False
+    
+    for i, line in enumerate(lines):
+        if not line:  # Skip empty lines
+            continue
+            
+        if reading_data:
+            data_list.append(line)
+        elif line.startswith('#'):
+            # Store the next non-empty line as the header value
+            key = line[1:]  # Remove the '#'
+            for next_line in lines[i+1:]:
+                if next_line and not next_line.startswith('#'):
+                    headers[key] = next_line
+                    break
+        
+        if line == '#GRID':
+            reading_data = True
+            
+    return data_list, headers
+
+def read_gxf(infile: str) -> Tuple[np.ndarray, Dict[str, Any]]:
+    """
+    Read a GXF file and return the grid data and metadata.
+    Following official GXF specifications from Geosoft.
+    
+    Parameters:
+    infile (str): Path to the GXF file
+    
+    Returns:
+    tuple: (grid_array, metadata)
+        - grid_array: numpy array containing the properly oriented grid
+        - metadata: dictionary containing all GXF parameters
+    """
+    # Get raw data and headers
+    data_list, headers = read_gxf_raw(infile)
+    
+    # Parse metadata with proper type conversion
+    metadata: Dict[str, Any] = {}
+    for key, value in headers.items():
+        try:
+            # Try converting to float first (handles scientific notation)
+            metadata[key] = float(value)
+            # If it's a whole number, convert to int
+            if metadata[key].is_integer():
+                metadata[key] = int(metadata[key])
+        except ValueError:
+            # If conversion fails, keep as string
+            metadata[key] = value.strip()
+    
+    # Convert data strings to numpy array
+    data_1d = np.array([])
+    for line in data_list:
+        # Handle both space and tab-delimited data
+        values = np.fromstring(line, sep=' ')
+        data_1d = np.concatenate((data_1d, values))
+    
+    # Get grid dimensions
+    nrows = int(headers['ROWS'])
+    ncols = int(headers['POINTS'])
```

Comment:
In lines https://github.com/fatiando/harmonica/pull/539/files#diff-5ac31ec9d8d11b47ff39f59bf25e1cad43484857580dd7b6c65cddd8e0f504cfR219-R229 we populate the `metadata` dictionary with parsed values (float, int, stripped str). Isn't it better to use `metadata` to get the number of rows and columns, instead of having to parse them again as ints?

That being said, maybe we want the keys in `metadata` to be lowercase (as you do in following lines with `nx`, `ny`, `x_inc`, etc.).